### PR TITLE
feat: returns the index of the core with the largest value

### PR DESCRIPTION
**Pull request by kobayu858**
## Description

Used to obtain core id at preemptive.

## Related links

None.

## Pre-review checklist for the PR author

- [x] I've confirmed the [self-checklist](https://docs.google.com/spreadsheets/d/1p5KMcm752iCySHJz0xL1qf6fcLDKprbo/edit#gid=863618628).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: lib/src/homogeneous.rs [hunk 1]
```diff
@@ -41,6 +41,19 @@ impl ProcessorBase for HomogeneousProcessor {
...
+        self.cores
+            .iter()
+            .enumerate()
+            .filter_map(|(index, core)| {
+                core.processing_node()
+                    .as_ref()
+                    .and_then(|node_data| node_data.params.get(key).map(|&value| (index, value)))
+            })
+            .max_by_key(|&(_, value)| value)
+            .map(|(index, _)| index)
```
Comment (atsushi421) [inline lib/src/homogeneous.rs:55]: Fixed a few clarifications.

```suggestion
        self.cores.iter()
    .enumerate()
    .filter_map(|(index, core)| {
        let node_data = core.processing_node().as_ref()?;
        let value = node_data.params.get(key)?;
        Some((index, *value))
    })
    .max_by_key(|&(_, value)| value)
    .map(|(index, _)| index)
```

File: lib/src/core.rs [hunk 2]
```diff
@@ -10,9 +11,11 @@ pub enum ProcessResult {
     Done(NodeData),
 }
 
-#[derive(Clone)]
+#[derive(Clone, Getters, Debug)]
 pub struct Core {
+    #[get = "pub"]
```
Comment (atsushi421) [inline lib/src/core.rs:16]:
```suggestion
    #[get_copy = "pub with_prefix"]
```

File: lib/src/core.rs [hunk 3]
```diff
@@ -10,9 +11,11 @@ pub enum ProcessResult {
...
     pub is_idle: bool,
+    #[get = "pub"]
```
Comment (atsushi421) [inline lib/src/core.rs:18]:
```suggestion
    #[get_copy = "pub with_prefix"]
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/homogeneous.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/homogeneous.rs:55]: Changed.

File: lib/src/core.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/core.rs:16]: I have never know about this before. Thank you very much.

File: lib/src/core.rs (hunk 3 above)
Comment (kobayu858) [reply lib/src/core.rs:18]: Changed.

Comment (kobayu858) [reply lib/src/core.rs]: If you need to copy, implement a method to return clone.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Pull request merged**